integ-test/src/test/java/org/opensearch/sql/ppl/HeadCommandIT.java

```diff
@@ -60,6 +61,30 @@ public void testHeadWithNumber() throws IOException {
         rows("Nanette", 28));
   }
 
+  @Test
+  public void testHeadWithLargeNumber() throws IOException {
+    setMaxResultWindow(TEST_INDEX_ACCOUNT, 10);
+    JSONObject result =
+        executeQuery(String.format(
+            "source=%s | fields firstname, age | head 15", TEST_INDEX_ACCOUNT));
```

Comment:
should we add more test? e.g. querySize < head limit, it should still return 15 records.

Reply:
I don't think I understand. Could you elaborate?

Reply:
https://github.com/opensearch-project/sql/issues/703#issuecomment-1211422130